src/Compilers/CSharp/Test/Emit/Attributes/AttributeTests.cs

```diff
@@ -7995,16 +8069,141 @@ public void GenericAttributeType()
+                // public class B<T> : Attribute // 17
+                Diagnostic(ErrorCode.ERR_FeatureNotAvailableInVersion9, "Attribute").WithArguments("generic attributes", "10.0").WithLocation(22, 21),
+                // (26,24): error CS8773: Feature 'generic attributes' is not available in C# 9.0. Please use language version 10.0 or greater.
+                // public class C<T, U> : Attribute // 18
+                Diagnostic(ErrorCode.ERR_FeatureNotAvailableInVersion9, "Attribute").WithArguments("generic attributes", "10.0").WithLocation(26, 24));
+
+            comp = CreateCompilation(source);
+            comp.VerifyDiagnostics(
+                // (4,2): error CS0308: The non-generic type 'A' cannot be used with type arguments
+                // [A<>] // 1, 2
+                Diagnostic(ErrorCode.ERR_HasNoTypeVars, "A<>").WithArguments("A", "type").WithLocation(4, 2),
+                // (5,2): error CS0308: The non-generic type 'A' cannot be used with type arguments
+                // [A<int>] // 3, 4
+                Diagnostic(ErrorCode.ERR_HasNoTypeVars, "A<int>").WithArguments("A", "type").WithLocation(5, 2),
+                // (6,2): error CS0305: Using the generic type 'B<T>' requires 1 type arguments
+                // [B] // 5
+                Diagnostic(ErrorCode.ERR_BadArity, "B").WithArguments("B<T>", "type", "1").WithLocation(6, 2),
+                // (7,2): error CS7003: Unexpected use of an unbound generic name
+                // [B<>] // 6, 7
+                Diagnostic(ErrorCode.ERR_UnexpectedUnboundGenericName, "B<>").WithLocation(7, 2),
+                // (9,2): error CS0305: Using the generic type 'C<T, U>' requires 2 type arguments
+                // [C] // 9
+                Diagnostic(ErrorCode.ERR_BadArity, "C").WithArguments("C<T, U>", "type", "2").WithLocation(9, 2),
+                // (10,2): error CS0305: Using the generic type 'C<T, U>' requires 2 type arguments
+                // [C<>] // 10, 11
+                Diagnostic(ErrorCode.ERR_BadArity, "C<>").WithArguments("C<T, U>", "type", "2").WithLocation(10, 2),
+                // (11,2): error CS0305: Using the generic type 'C<T, U>' requires 2 type arguments
+                // [C<int>] // 12, 13
+                Diagnostic(ErrorCode.ERR_BadArity, "C<int>").WithArguments("C<T, U>", "type", "2").WithLocation(11, 2),
+                // (12,2): error CS7003: Unexpected use of an unbound generic name
+                // [C<,>] // 14, 15
+                Diagnostic(ErrorCode.ERR_UnexpectedUnboundGenericName, "C<,>").WithLocation(12, 2));
+        }
+
+        [Fact]
+        public void GenericAttributeTypeWithGenericAttributeFeature()
+        {
+            var source = @"
+using System;
+using System.Collections.Generic;
+
 [A<>]
 [A<int>]
 [B]
 [B<>]
 [B<int>]
+[B<List<int>>]
 [C]
 [C<>]
 [C<int>]
 [C<,>]
 [C<int, int>]
```

Comment:
It looks like `[A<>]`, `[A<int>]`, `[B]`, `[B<>]`, `[B<int>]`, `[C]`, `[C<>]`, `[C<int>]`, `[C<,>]`, `[C<int, int>]` are all covered in the previous test.

Reply:
Thanks. Will merge the tests.